snowflake/data_access_sync_from_target.go

```diff
@@ -36,52 +127,62 @@ func (s *AccessSyncer) importAllRolesOnAccountLevel(accessProviderHandler wrappe
-	roleEntities, err := repo.GetAccountRoles()
+	roleEntities, err := s.repo.GetAccountRoles()
 	if err != nil {
 		return err
 	}
 
+	wp := workerpool.New(getWorkerPoolSize(s.configMap))
+
 	for _, roleEntity := range roleEntities {
-		if _, exclude := excludedRoles[roleEntity.Name]; exclude {
+		if _, exclude := s.excludedRoles[roleEntity.Name]; exclude {
 			logger.Info("Skipping SnowFlake ROLE " + roleEntity.Name)
 			continue
 		}
 
-		err = s.transformAccountRoleToAccessProvider(roleEntity, processedAps, linkToExternalIdentityStoreGroups, availableTags, externalGroupOwners, shares, repo)
-		if err != nil {
-			logger.Warn(fmt.Sprintf("Error importing SnowFlake role %q: %s"+roleEntity.Name, err.Error()))
-		}
+		wp.Submit(func() {
+			err2 := s.transformAccountRoleToAccessProvider(roleEntity, processedAps, availableTags)
+			if err2 != nil {
+				logger.Warn(fmt.Sprintf("Error importing SnowFlake role %q: %s", roleEntity.Name, err2.Error()))
+			}
+		})
 	}
 
+	wp.StopWait()
+
 	err = accessProviderHandler.AddAccessProviders(values(processedAps)...)
 	if err != nil {
 		return fmt.Errorf("error adding access provider to import file: %s", err.Error())
 	}
 
 	return nil
 }
-func (s *AccessSyncer) shouldRetrieveTags(configMap *config.ConfigMap) bool {
-	standard := configMap.GetBoolWithDefault(SfStandardEdition, false)
-	skipTags := configMap.GetBoolWithDefault(SfSkipTags, false)
+
+func (s *AccessFromTargetSyncer) shouldRetrieveTags() bool {
+	standard := s.configMap.GetBoolWithDefault(SfStandardEdition, false)
+	skipTags := s.configMap.GetBoolWithDefault(SfSkipTags, false)
 
 	tagSupportEnabled := !standard && !skipTags
 
 	return tagSupportEnabled
 }
 
-func (s *AccessSyncer) transformAccountRoleToAccessProvider(roleEntity RoleEntity, processedAps map[string]*exporter.AccessProvider, linkToExternalIdentityStoreGroups bool, availableTags map[string][]*tag.Tag, externalGroupOwners string, shares []string, repo dataAccessRepository) error {
+func (s *AccessFromTargetSyncer) transformAccountRoleToAccessProvider(roleEntity RoleEntity, processedAps map[string]*exporter.AccessProvider, availableTags map[string][]*tag.Tag) error {
 	logger.Info(fmt.Sprintf("Reading SnowFlake ROLE %s", roleEntity.Name))
 
 	roleName := roleEntity.Name
 	externalId := roleName
 	currentApType := ptr.String(access_provider.Role)
-	fromExternalIS := s.comesFromExternalIdentityStore(roleEntity, externalGroupOwners)
+	fromExternalIS := s.comesFromExternalIdentityStore(roleEntity, s.externalGroupOwners)
 
-	users, groups, accessProviders, err := s.retrieveWhoEntitiesForRole(roleEntity, externalId, currentApType, fromExternalIS, linkToExternalIdentityStoreGroups, repo)
+	users, groups, accessProviders, err := s.retrieveWhoEntitiesForRole(roleEntity, externalId, currentApType, fromExternalIS)
 	if err != nil {
 		return err
 	}
 
+	// Locking to make sure only one goroutine can read & write to the processedAps map at a time
+	s.lock.Lock()
```

Comment:
Maybe we should replace the map with a `sync/map` instead of managing the mutex ourself?